Commands/Test.js

```diff
@@ -18,121 +21,86 @@ program
 	.option('-f, --force', 'Force rebuild applications.')
 	.parse(process.argv);
 
-// Logging option for Output helper
-global.logging = program.logging;
 // The root of the project
 global.projRoot = path.join(__dirname, '..');
 // Get the Appium settings from the input flags
-global.server = {
-	host: program.address,
-	port: program.port
-};
 global.appcCLI = program.cli;
+
 global.appcSDK = program.sdk;
 
-// Setup the logging directory for this run
-Output.setupLogDir(err => {
-	if (err) {
-		console.error(`An error occured setting up the logging directory!: ${err}`);
+// Validate that the passed platforms passed are valid
+const
+	platforms = program.platforms.split(','),
+	suppPlatforms = [ 'ios', 'android' ];
+
+platforms.forEach(platform => {
+	if (!suppPlatforms.includes(platform)) {
+		tiapp.error(`'${platform}' is not a valid platform.`);
 		process.exit();
 	}
 });
 
-// Set the global for the hostOS to the current OS being run on
+// Set the host value to the current OS being run on
+let host;
+
 switch (process.platform) {
 	case 'darwin':
 		global.hostOS = 'Mac';
+		host = 'Mac';
 		break;
 
 	case 'win32':
 		global.hostOS = 'Windows';
+		host = 'Windows';
 		break;
+
+	default:
+		throw Error('Running on an unsupported OS');
 }
 
-// TODO: Cater for new SIGINT requirements
-// If the process is killed in the console, force close all test devices
-process.on('SIGINT', () => {
-	Device.quickKill();
-	process.exit();
-});
+const appcConf = {
+	username: appc.user,
+	password: appc.pass,
+	organisation: appc.org,
+	cli: program.cli,
+	sdk: program.sdk
+};
 
-// Validate that the platforms passed are valid
-let
-	platforms = {},
-	suppPlatforms = [ 'iOS', 'Android' ];
+async function run() {
+	try {
+		await tiapp.appcSetup(appcConf, 'production');
+		await tiapp.startAppium(global.projRoot, { hostname: program.address, port: program.port });
+	} catch (err) {
+		tiapp.error(err);
```

Comment:
If credentials aren't defined or the server doesn't start we're in for a bad time. I'd add a process.exit(1) after printing the error here to prevent a runaway series of issues. (I know this was my code, I'll be making the same change to phoenix)